File: multipost/code/.multipost/flutter_app/multipost/lib/domain/entities/post.dart
class Post {
  final String id;
  final String title;
  final String description;
  final List<String> images;
  final List<String> channels;
  final List<String> platforms;
  final DateTime? scheduledAt;
  final bool published;

  Post({
    required this.id,
    required this.title,
    required this.description,
    required this.images,
    required this.channels,
    required this.platforms,
    this.scheduledAt,
    required this.published,
  });

  factory Post.fromJson(Map<String, dynamic> json) {
    return Post(
      id: json['id'],
      title: json['title'],
      description: json['description'],
      images: List<String>.from(json['images'].map((img) => img['data'])),
      channels: List<String>.from(json['channels']),
      platforms: List<String>.from(json['platforms']),
      scheduledAt: json['scheduled_at'] != null
          ? DateTime.parse(json['scheduled_at'])
          : null,
      published: json['published'],
    );
  }
}

File: multipost/code/.multipost/go_app/multipost-go/internal/models/post.go
package models

import (
	"time"
)

type Post struct {
	ID          string     `json:"id" gorm:"primaryKey;type:text"`
	Title       string     `json:"title"`
	Description string     `json:"description"`
	Images      []Image    `json:"images"`
	Socials     []Link     `json:"socials" gorm:"many2many:post_links;foreignKey:ID;joinForeignKey:PostID;References:ID;joinReferences:LinkID;constraint:OnDelete:CASCADE"`
	ScheduledAt *time.Time `json:"scheduled_at"`
	Published   bool       `json:"published"`
	UserID      string     `json:"user_id"`
	CreatedAt   time.Time  `json:"created_at"`
	UpdatedAt   time.Time  `json:"updated_at"`
	Platform    string     `json:"platform"` // Новое поле (telegram, vk, reddit)
}
